src/test/librados/misc.cc

```diff
@@ -510,6 +510,29 @@ TEST_F(LibRadosMiscPP, AssertExistsPP) {
   ASSERT_EQ(-EEXIST, ioctx.create("asdffoo", true));
 }
 
+TEST_F(LibRadosMiscPP, AssertVersionPP) {
+  char buf[64];
+  memset(buf, 0xcc, sizeof(buf));
+  bufferlist bl;
+  bl.append(buf, sizeof(buf));
+
+  ASSERT_EQ(0, ioctx.create("asdfbar", true));
+  ASSERT_EQ(0, ioctx.write_full("asdfbar", bl));
```

Comment:
That was intentional. I write to the object twice in order to guarantee that its version number is at least 1 (not 0) so that I can decrement it afterwards (it's unsigned). But I see now that both writes are in the same op so it does not actually have the intended effect. I will rework it.